cvat-core/src/server-proxy.js

```diff
@@ -109,70 +105,42 @@
-                    if (response.headers['set-cookie']) {
-                        // Browser itself setup cookie and header is none
-                        // In NodeJS we need do it manually
-                        let cookies = '';
-                        for (let cookie of response.headers['set-cookie']) {
-                            [cookie] = cookie.split(';'); // truncate extra information
-                            const name = cookie.split('=')[0];
-                            const value = cookie.split('=')[1];
-                            if (name === 'csrftoken') {
-                                setCSRFHeader(value);
-                            }
-                            Cookie.set(name, value);
-                            cookies += `${cookie};`;
-                        }
-
-                        Axios.defaults.headers.common.Cookie = cookies;
-                    } else {
-                        // Browser code. We need set additinal header for authentification
-                        let csrftoken = response.data.csrf;
-                        if (csrftoken) {
-                            setCSRFHeader(csrftoken);
-                            Cookie.set('csrftoken', csrftoken);
-                        } else {
-                            csrftoken = Cookie.get('csrftoken');
-                            if (csrftoken) {
-                                setCSRFHeader(csrftoken);
-                            } else {
-                                throw new ScriptingError(
-                                    'An environment has been detected as a browser'
-                                    + ', but CSRF token has not been found in cookies',
-                                );
-                            }
-                        }
-                    }
-                }
-
-                const host = config.backendAPI.slice(0, -7);
-                let csrf = null;
+            async function register(username, firstName, lastName, email, password1, password2) {
+                let response = null;
                 try {
-                    csrf = await Axios.get(`${host}/auth/csrf`, {
+                    const data = JSON.stringify({
+                        username,
+                        first_name: firstName,
+                        last_name: lastName,
+                        email,
+                        password1,
+                        password2,
+                    });
+                    response = await Axios.post(`${config.backendAPI}/auth/register`, data, {
                         proxy: config.proxy,
                     });
                 } catch (errorData) {
                     const code = errorData.response ? errorData.response.status : errorData.code;
                     throw new ServerError(
-                        'Could not get CSRF token from a server',
+                        'Could not register `{0}` user on the server'.format(username),
```

Comment:
Is the method format() work? 
JS standard doesn't have such a function as far as I know. Here we should use template strings.